src/dune_rules/ocaml_flags.ml

```diff
@@ -167,9 +167,31 @@ let map_common t ~f =
 
 let append_common t flags = map_common t ~f:(fun l -> l @ flags)
 
-let with_vendored_warnings t = append_common t vendored_warnings
```

Comment:
I'm a bit surprised that it's necessary to remove the warning options from the CLI. Isn't `-w <whatever> -w -a` equivalent to `-w -a`?

Reply:
I've forgotten my exact thoughts here, but one reason to do this is so that vendored libraries do not get affected by warning changes in the workspace.

Reply:
We should address Etienne's question here before we consider this for 3.8